jobs/bump-jenkins-plugins.Jenkinsfile

```
@@ -0,0 +1,125 @@
+fork_repo = "coreosbot-releng/fedora-coreos-pipeline"
+botCreds = "github-coreosbot-releng-token-username-password"
+pr_branch = "pluginsupdate"
+
+def getVersionFromPluginUrl(pluginUrl) {
+    //example url : https://updates.jenkins.io/download/plugins/${pluginName}/latest/${pluginName}.hpi
+    def parts = pluginUrl.split("/")
+    def pluginVersion
+    if (parts.size() >= 4) {
+        def groupId = parts[-3]
+        pluginVersion = parts[-2]
+    } else {
+        error("Unable to extract plugin version from the URL.")
+    }
+    return pluginVersion
+}
+
+node {
+    checkout scm: [
+        $class: 'GitSCM',
+        branches: [[name: '*/pluginsupdate']],
+        userRemoteConfigs: [[url: "https://github.com/${fork_repo}.git"]],
+        extensions: [[$class: 'WipeWorkspace']]
+    ]
+    // these are script global vars
+    pipeutils = load("utils.groovy")
+
+    properties([
+        pipelineTriggers([
+            // check once a month
+            pollSCM('H H 1 * *')
+        ]),
+        buildDiscarder(logRotator(
+            numToKeepStr: '100',
+            artifactNumToKeepStr: '100'
+        )),
+        durabilityHint('PERFORMANCE_OPTIMIZED')
+    ])
+
+    try {
+        shwrap("""
+            git config --global user.name "CoreOS Bot"
+            git config --global user.email "coreosbot-releng@fedoraproject.org"
+        """)
+
+        def pluginslist
+        def pluginsToUpdate = [:]
+        def plugins_lockfile = "jenkins/controller/plugins.txt"
+
+        stage("Read plugins.txt") {
+            shwrapCapture("""
+                git clone --branch pluginsupdate https://github.com/${fork_repo}.git
```

wonder if we should make this a shallow clone? do we need git history?

```suggestion
                git clone --depth=1 --branch pluginsupdate https://github.com/${fork_repo}.git
```

---

Shallow cloning can be a good optimization, especially if we don't need the entire history. It could spped up the cloning process too.